api/base/utils.py

```diff
@@ -19,17 +20,24 @@ def absolute_reverse(view_name, query_kwargs=None, args=None, kwargs=None):
     return url
 
 
-def get_object_or_404(model_cls, query_or_pk):
+def get_object_or_error(model_cls, query_or_pk):
+
     if isinstance(query_or_pk, basestring):
         query = Q('_id', 'eq', query_or_pk)
     else:
         query = query_or_pk
     try:
-        return model_cls.find_one(query)
+        obj = model_cls.find_one(query)
+
+        if getattr(obj, 'is_deleted', False) is True:
```

Comment:
`try` blocks should be limited to the code that can raise the expected error(s)--in this case, the `find_one` call. All this other code does not need to be in the `try` block.